src/controller/community.ts

```diff
@@ -20,10 +20,17 @@ export default class CommunityController extends BaseController {
 
 	async handleGetIndex(_req: Request, res: Response, _next: NextFunction) {
 		try {
+			const { user } = res.locals;
 			const posts = (await this.postRepository.getAllPostsByVerifiedAgencies()).sort(
 				(a, b) => a.createdAt.getTime() - b.createdAt.getTime(),
 			);
-			this.renderView(res, 'pages/community', { posts });
+
+			if (user?.userRole !== 'partner') {
+				return this.renderView(res, 'pages/community', { posts });
+			}
+
```

Comment:
Can you add a condition so that if agency.isVerified it returns {posts, agency} and else {posts} even for user.userRole == 'partner'

Reply:
I can but its better this way. If its not a partner agency is not necessary and we have to check if its verified on the frontend anyway.